src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Specific.cs

```diff
@@ -77,6 +79,147 @@ public void ConnectWithConflictingDirections_Throws_UnauthorizedAccessException(
         }
 
         [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        public async Task MultipleWaitingClients_ServerServesOneAtATime(int numClients)
+        {
+            string name = GetUniquePipeName();
+            using (NamedPipeServerStream server = new NamedPipeServerStream(name))
+            {
+                var clients = new List<NamedPipeClientStream>(numClients);
+                var clientConnects = new List<Task>();
+                for (int i = 0; i < numClients; i++)
+                {
+                    clients.Add(new NamedPipeClientStream(name));
+                    clientConnects.Add(clients[i].ConnectAsync());
+                }
+
+                while (clientConnects.Count > 0)
+                {
+                    Task serverWait = server.WaitForConnectionAsync();
+
+                    int connectedIndex = clientConnects.IndexOf(await Task.WhenAny(clientConnects));
+                    NamedPipeClientStream client = clients[connectedIndex];
+                    clientConnects.RemoveAt(connectedIndex);
+                    clients.RemoveAt(connectedIndex);
+
+                    await serverWait;
+
+                    Task writeAsync = client.WriteAsync(new byte[1], 0, 1);
+                    Task<int> readAsync = server.ReadAsync(new byte[1], 0, 1);
+                    await Task.WhenAll(writeAsync, readAsync);
+                    Assert.Equal(1, await readAsync);
+
+                    writeAsync = server.WriteAsync(new byte[1], 0, 1);
+                    readAsync = client.ReadAsync(new byte[1], 0, 1);
+                    await Task.WhenAll(writeAsync, readAsync);
+                    Assert.Equal(1, await readAsync);
+
+                    server.Disconnect();
+                }
+            }
+        }
+
+        [Fact]
+        public void MaxNumberOfServerInstances_TooManyServers_Throws()
```

Comment:
This test only made sense to me in light of the comments at `SharedServer.Get` perhaps a brief mention here can be helpful.

Reply:
Ok, will add some comments.

Reply:
Added

Reply:
Thanks.